src/plugins/data_source_management/public/plugin.ts

```diff
@@ -303,10 +319,85 @@ export class DataSourceManagementPlugin
     };
     setRenderAssociatedObjectsDetailsFlyout(renderAssociatedObjectsDetailsFlyout);
 
+    const getDirectQuerySyncFromUrl = (): boolean | null => {
+      const parsedHash = parseUrlHash(window.location.href);
+      if (parsedHash && parsedHash.query) {
+        const directQuerySync = parsedHash.query.directQuerySync;
+        if (directQuerySync === 'true') return true;
+        if (directQuerySync === 'false') return false;
+      }
+      return null;
+    };
+
+    const shouldMountBanner = (): boolean => {
+      const urlFlag = getDirectQuerySyncFromUrl();
+      if (urlFlag !== null) {
+        return urlFlag;
+      }
+      return this.config.dashboardDirectQuerySyncEnabled;
+    };
+
+    const updateBanner = () => {
+      const appId = this.currentAppId;
+
+      if (appId === 'dashboards') {
+        const hash = window.location.hash;
+        const isDashboardViewMatch = hash.match(/#\/view\/([^\/?]+)(\?.*)?$/);
```

Comment:
you might be able to use the application utils https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/public/application/utils/parse_app_url.ts#L49

Reply:
theres other url utils too if it doesn't fit your needs

Reply:
oh i see you export the those helpers. should we use it here then?

Reply:
nice catch, I have also applied the `parseUrlHash()` to here. In addition, I'm actually discovered a cleaner way to get the hash + id by just checking `parsedHash.pathname` it should be something like `/view/411eaf12-3e8f-4e70-b8d9-c1f2f00178e2` which is exactly what we need from the url.